Sming/third-party/.patches/umm_malloc.patch

```diff
@@ -16,12 +16,12 @@ index fcc4fb1..661b94a 100644
 +
 +#ifdef __ets__
 +// ESP specific code
++#include "debug_progmem.h"
 +extern void ets_intr_lock();
 +extern void ets_intr_unlock();
-+extern void ets_printf(const char*, ...);
 +
-+#define printf ets_printf
-+#define puts(...) ets_printf("%s",...)
++#define printf debug_i
```

Comment:
I would be extremely cautious in adding DRAM magic to code related to such an important stuff as low level memory allocation. Was that change tested extensively on a device with a lot of memory allocation / freeing operations? Is there a slow down? Is the code functioning without segfaults?